framework/src/base/Assembly.C

```diff
@@ -939,7 +943,12 @@ Assembly::prepare()
     unsigned int vi = ivar.number();
     unsigned int vj = jvar.number();
 
-    jacobianBlock(vi, vj).resize(ivar.dofIndices().size(), jvar.dofIndices().size());
+    AllLocalDofIndicesThread aldit(_sys.system(), {jvar.name()});
```

Comment:
Okay, I will have to define it somewhere else and store in a vector so that I can extract the dofs depending on jvar when needed.